Bpk 3813 dev android create bar chart component (#440)
Skyscanner/backpack-android: docs/BarChart/README.md

# Bar Chart

## Installation

Backpack Android is available through [Jitpack](https://jitpack.io/#Skyscanner/backpack-android). Check the main [Readme](https://github.com/skyscanner/backpack-android#installation) for a complete installation guide.

## Usage

> The Bar Chart component can be used in both XML and Kotlin/Java

Reviewer: Since it can't really be used only in XML I think we should change this to clarify it a bit.

E.g. The Bar Chart component can be used in both XML and Kotlin/Java, however, feeding the component with data can only be done programatically.

Author: Done!

Example of a Bar Chart in XML

```xml
<net.skyscanner.backpack.barchart.BpkBarChart
    android:id="@+id/bar_chart"
    android:layout_width="match_parent"
    android:layout_height="240dp" />
```

Example of a Bar Chart in Kotlin

```Kotlin
import net.skyscanner.backpack.barchart.BpkBarChart

BpkBarChart(context).apply {
    model = //.. initialisation
}
```

> To render the data you need to set a view model to Bar Chart in code.
> Detailed documentation on each property of the model is available in Javadoc.
> 
> Here's some basic example in Kotlin:

Reviewer: ```suggestion
To render data you need to set a view model to `BarChart` in code.
Detailed documentation on each property of the model is available [here](https://backpack.github.io/android/versions/latest/net.skyscanner.backpack.barchart).

Here's some basic example in Kotlin:
```

```kotlin

fun createGroup() = BpkBarChart.Group(
  title = "Group Title",
  items = mutableListOf<BpkBarChart.Column>().apply {
    for (i in 0 until 10) {
      add(BpkBarChart.Column(
        title = "Bar Title",
        subtitle = "Bar Subtitle",
        badge = "Bar Badge",
        value = 0.5f,
        inactive = false
      ))
    }
  }
)

> barChart.model = BpkBarChart.Model(
>         groups = listOf(
>           createGroup(),
>           createGroup()
>         ),
>         legend = BpkBarChart.Legend(
>           "£",
>           "No Price"
>         )
>       )

Reviewer: ```suggestion
barChart.model = BpkBarChart.Model(
  groups = listOf(
    createGroup(),
    createGroup()
  ),
  legend = BpkBarChart.Legend(
    "£",
     "No Price"
   )
 )
```

```

Another example can be found in `BarChartStory` in the sample project.

## Theme Props

> All the colours support `ColorStateList`.

Reviewer: ```suggestion
All colours support `ColorStateList`.
```
